tests/pgsmo_tests/utils.py

```diff
@@ -106,29 +108,30 @@ def _execute(self, query, params=None):
         self._has_been_read = True
 
 
-class MockConnection(connection):
+class MockServerConnection(PostgreSQLConnection):
+    '''Class used to mock ServerConnection objects for testing'''
     def __init__(
-            self,
-            cur: Optional[MockCursor],
-            version: str = '90602',
-            name: str = 'postgres',
-            host: str = 'localhost',
-            port: str = '25565',
-            user: str = 'postgres'):
-        # Setup the properties
-        self._server_version = version
+        self,
+        cur: Optional[MockCursor] = None,
+        connection: Optional[MockPsycopgConnection] = None,
+        version: str = '90602',
+        name: str = 'postgres',
+        host: str = 'localhost',
+        port: str = '25565',
+        user: str = 'postgres'):
 
         # Setup mocks for the connection
         self.close = mock.MagicMock()
         self.cursor = mock.MagicMock(return_value=cur)
```

Comment:
Don't we neet to check if cur is null when we use it here: self.cursor = mock.MagicMock(return_value=cur)

Reply:
I believe tests that need to call cursor method pass in a mock cursor. If they don't pass one in, the cursor method would simply return None. I think this is fine, but do you not want this behavior?

Reply:
I was wondering if it throws an error or complains about it. If it does not, then it should be okay as we will pass it when needed

Reply:
Nope, cur defaults to None here, which is a valid return_value.